pkg/client/testing/core/fixture.go

```diff
@@ -286,6 +297,15 @@ func (t *tracker) add(obj runtime.Object, replaceExisting bool) error {
 			return err
 		}
 
+		// Propagate namespace to the new object if hasn't already been set.
+		if len(newMeta.GetNamespace()) == 0 {
+			newMeta.SetNamespace(ns)
+		}
+
+		if newMeta.GetNamespace() != ns {
```

Comment:
This is kind of awkward for what the tracker is supposed to do.  There are many tests that could span namespaces, so erroring out here seems incorrect.